test/sshkit/ssh_test.exs

```diff
@@ -13,36 +13,150 @@ defmodule SSHKit.SSHTest do
+        options:     [user_interaction: false, user: String.to_charlist(user)],
+        port:        22,
+        ref:         :sandbox,
+        ssh_modules: Keyword.get(options, :ssh_modules)
+      }
+
+      assert connect(state[:host], options) == {:ok, conn}
+      refute_received :closed_sandbox_connection
+    end
+
+    @options [ssh_modules: %{ssh: SSHSandboxError, ssh_connection: :ssh_connection}]
+    test "return error and do not attempt to close if connection cannot be opened", state do
+      assert connect(state[:host], @options) == {:error, :sandbox}
+      refute_received :closed_sandbox_connection
+    end
+
+    test "return error and do not attempt to close if no host given" do
+      assert connect(nil, @options) == {:error, "No host given."}
+      refute_received :closed_sandbox_connection
+    end
+
+    test "error if options not provided as List", state do
+      options = %{user: "me", password: "secret"}
+      assert_raise FunctionClauseError, fn -> connect(state[:host], options) end
+    end
   end
 
   describe "connect/3" do
     @options [ssh_modules: %{ssh: SSHSandboxSuccess, ssh_connection: :ssh_connection}]
-    test "execute function on open connection" do
-      host = "foo"
+    test "execute function on open connection", state do
       func = fn(conn) ->
         assert conn.ssh_modules == Keyword.get(@options, :ssh_modules)
         42
       end
 
-      assert connect(host, @options, func) == {:ok, 42}
+      assert connect(state[:host], @options, func) == {:ok, 42}
       assert_received :closed_sandbox_connection
     end
 
-    test "close connection although function errored" do
-      host = "foo"
+    test "close connection although function errored", state do
       func = fn(_conn) -> raise("error") end
 
-      assert_raise RuntimeError, "error", fn -> connect(host, @options, func) end
+      assert_raise RuntimeError, "error", fn -> connect(state[:host], @options, func) end
       assert_received :closed_sandbox_connection
     end
 
     @options [ssh_modules: %{ssh: SSHSandboxError, ssh_connection: :ssh_connection}]
-    test "error during connect" do
-      host = "foo"
+    test "error during connect", state do
       func = fn(_conn) -> flunk "should never be called" end
 
-      assert connect(host, @options, func) == {:error, :sandbox}
+      assert connect(state[:host], @options, func) == {:error, :sandbox}
       refute_received :closed_sandbox_connection
     end
   end
+
+  describe "close/1" do
+    @options [ssh_modules: %{ssh: SSHSandboxSuccess, ssh_connection: :ssh_connection}]
+    test "call close on the connection" do
+      conn = %SSHKit.SSH.Connection{
+        host:        'test',
+        options:     [user_interaction: false],
+        port:        22,
+        ref:         :sandbox,
+        ssh_modules: Keyword.get(@options, :ssh_modules)
+      }
+
+      assert close(conn) == :ok
+      assert_received :closed_sandbox_connection
+    end
```

Comment:
It uses `Connection.close/1` under the hood (which finally uses Erlang's `:ssh.close/1`).
In the [Erlang documentation](http://erlang.org/doc/man/ssh_connection.html#close-2) *(if I'm able to read it correctly...)*  it says it will always return `:ok`.